lib/AST/RawComment.cpp

```diff
@@ -247,3 +247,13 @@ StringRef Decl::getBriefComment() const {
   Context.setBriefComment(this, Result);
   return Result;
 }
+
+CharSourceRange RawComment::getCharSourceRange(const SourceManager &SM) {
+  if (this->isEmpty()) {
+    return CharSourceRange();
+  }
+
+  auto Start = this->Comments.front().Range.getStart();
+  auto End = this->Comments.back().Range.getEnd();
+  return CharSourceRange(SM, Start, End);
```

Comment:
We actually don't need `SM` here, it's reasonable to assume `Start` and `End` are well formed. so something like:
```c++
  return CharSourceRange(Start, (char*)End.getOpaquePointerValue() -
    (char*)Start.getOpaquePointerValue());
```